app/Http/Controllers/KeyFobController.php

```diff
@@ -77,7 +85,12 @@ public function update($id)
      */
     public function destroy($id)
     {
-        $fob = KeyFob::findOrFail($id);
+        $fob = KeyFob::where('id', $id);
+
+        if(!\Auth::user()->isAdmin()){
+            $fob->where('user_id', \Auth::user()->id);
+        }
+        $fob->first();
         $fob->markLost();
```

Comment:
`$fob` will be an instance of `Illuminate\Database\Query\Builder` by line 94, so the `markLost` call will fail.

Also worth bringing back `firstOrFail` so we return a HTTP 404 if given an invalid key fob ID, otherwise we'll probably get a PHP fatal error "Call to a member function markLost() on null".

```suggestion
        $fobQuery = KeyFob::where('id', $id);

        if(!\Auth::user()->isAdmin()){
            $fobQuery->where('user_id', \Auth::user()->id);
        }
        $fob = $fobQuery->firstOrFail();
        $fob->markLost();
```

Reply:
great suggestion 👍